addons/survey/models/survey_survey.py

```diff
@@ -809,6 +817,29 @@ def _get_survey_questions(self, answer=None, page_id=None, question_id=None):
             questions = questions & answer.predefined_question_ids
         return questions, page_or_question_id
 
+    def get_correct_answers(self, question_ids=False):
+        """ Returns a dictionary linking the survey question ids with their correct answers.
+        Limited to some questions if the question_ids parameter is specified. """
```

Comment:
You can add here that it is the caller's reponsibility to check that answers can be given (users can't change their answers).